packages/angular_devkit/build_angular/src/browser/utils.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as browserslist from 'browserslist';
+import * as caniuse from 'caniuse-api';
+import * as ts from 'typescript';
+
+export function isDifferentialLoadingNeeded(
+  projectRoot: string,
+  target: ts.ScriptTarget): boolean {
+
+  const supportES2015 = target !== ts.ScriptTarget.ES3 && target !== ts.ScriptTarget.ES5;
+
+  return supportES2015 && isEs5SupportNeeded(projectRoot);
+}
+
+let _isEs5SupportNeeded: boolean | undefined = undefined;
+
+export function isEs5SupportNeeded(projectRoot: string): boolean {
+
+  if (typeof _isEs5SupportNeeded !== 'undefined') {
```

Comment:
This doesn't seem correct. The function will not always return the same value for the same file, because a value might be already set.

What is the reason behind this? Have you a performance hit?